### ISSUE TITLE
linter: JS plugins "rule.meta.fixable"

### ISSUE THREAD

**Issue by jackholden** [oxc-project/oxc#17967]
### What version of Oxlint are you using?

1.38.0

### What command did you run?

npx oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
    "jsPlugins": ["eslint-plugin-lit-a11y"],
    "rules": {
        "lit-a11y/alt-text": "warn"
    }
}
```

```jsonc
{
  "jsPlugins": ["@html-eslint/eslint-plugin"],
  "rules": {
    "html/no-duplicate-class": "error",
  }
}
```

### What happened?

I saw that JS plugins are in technical preview so wanted to see if I could get the `eslint-plugin-lit-a11y` working. 

I also tried `@html-eslint/eslint-plugin` though i wasn't expecting that to work due to the ".html" extension not being supported. 

I ended up with this output for both:

`TypeError: Invalid 'rule.meta.fixable'`

Am I doing something silly? This plugin doesn't support fixable so is set to null. Is null not supported yet?

Thanks in advance.

Full output:
```
Failed to parse oxlint configuration file.

  × Failed to load JS plugin: eslint-plugin-lit-a11y
  │   TypeError: Invalid `rule.meta.fixable`
  │   TypeError: Invalid `rule.meta.fixable`
  │     at registerPlugin (file:///Users/jack/Developer/Phase/astro/node_modules/oxlint/dist/lint.js:15217:63)
  │     at loadPlugin (file:///Users/jack/Developer/Phase/astro/node_modules/oxlint/dist/lint.js:15203:35)
```

```
Failed to parse oxlint configuration file.

  × Failed to load JS plugin: @html-eslint/eslint-plugin
  │   TypeError: Invalid `rule.meta.fixable`
  │     at registerPlugin (file:///Users/jack/Developer/Phase/astro/node_modules/oxlint/dist/lint.js:15217:63)
  │     at loadPlugin (file:///Users/jack/Developer/Phase/astro/node_modules/oxlint/dist/lint.js:15203:35)
```

Link to rule fixer definition:

https://github.com/open-wc/open-wc/blob/5aeaf35e01a2f15b6663d71102eeea3333e4c57d/packages/eslint-plugin-lit-a11y/lib/rules/aria-attrs.js#L27

**Comment by connorshea** [comment]
Hmm, `null` should be fine, I'm not sure why this isn't working. https://github.com/oxc-project/oxc/blob/efacb138be9473517e29b05ce14687aabec7a2ba/apps/oxlint/src-js/plugins/load.ts#L172

Is this a current version of the lit-a11y plugin or is it an older version?

**Comment by connorshea** [comment]
I can confirm this behavior after trying the plugin via https://github.com/Sysix/oxlint-jsplugin-checker/blob/3cccfe269f6f84dc6b395590603eb050ba002513/src/index.ts on oxlint 1.39.0. Weird, I have no idea why this is happening.

**Comment by connorshea** [comment]
I guess it uses eslint-rule-extender, which is the only reason I can see for it potentially having a weird fixable value? https://github.com/kaicataldo/eslint-rule-extender

It's doing something weird, I don't really understand it: https://github.com/open-wc/open-wc/blob/master/packages/eslint-plugin-lit-a11y/lib/utils/HasLitHtmlImportRuleExtension.js

**Comment by connorshea** [comment]
Yeah it's because of this rule extender thing: https://github.com/kaicataldo/eslint-rule-extender/blob/967e9b7f856c18398f1e48343c35364a5a09112a/lib/index.js#L21

This is `undefined || null || false` when `rule.meta.fixable` is `null` in the rule definition, and thus becomes `false`

`(undefined || null || false) != null` results in `true`, so we pass this `if`:

https://github.com/oxc-project/oxc/blob/efacb138be9473517e29b05ce14687aabec7a2ba/apps/oxlint/src-js/plugins/load.ts#L172

And then hit 

https://github.com/oxc-project/oxc/blob/efacb138be9473517e29b05ce14687aabec7a2ba/apps/oxlint/src-js/plugins/load.ts#L173

And boom, error. I have no idea why the rule extender library works this way, probably unintentional. But I guess ESLint itself must handle false and null the same? ¯\\\_(ツ)_/¯

**Comment by connorshea** [comment]
Yeah I guess ESLint's implementation _technically_ treats `false` the same as null, due to javascript nonsense, even though officially it should only allow `null`, `"code"`, or `"whitespace"`. (really it doesn't perform this check at all until the fix is considered for application, as opposed to when starting the program like oxlint does, so 🤷‍♂️ )

https://github.com/eslint/eslint/blob/23490b266276792896a0b7b43c49a1ce87bf8568/lib/linter/linter.js#L589-L593

**Comment by connorshea** [comment]
I think we will need to support `false` here, as a few other libraries also do it: https://github.com/keithamus/eslint-plugin-escompat/blob/main/lib/index.js#L18

**Comment by overlookmotel** [comment]
#18700 suggests we need to support `true` too!

**Comment by overlookmotel** [comment]
Sorry it's taken me a while to come back on this. I've been waving back and forwards on whether we should support or not. My conclusion is this:

Ultimately, we've made a trade-off. We've decided to align with ESLint. Upside is strong chance of high adoption, downside is that we get stuck with some of their (IMO) poor choices. Our selling point to users is "switch to Oxlint and everything just works like before, but faster". So unfortunately, to fulfil that promise, I do think we have to support `false` (and `true`).

`eslint-plugin-compat` is high on the list of most popular ESLint plugins, and @connorshea has discovered that it also has this problem. So if we don't support it, the impact is quite high for users.

The downside of supporting is very minimal. We just have to check for 2 more values, and that check only happens when registering plugins, not for every file that's linted. So the perf impact is basically zero.

So... in short... while I do see the point of a principled stand on strictly validating input, is this really the hill that we want to die on?

cc @camc314

**Comment by connorshea** [comment]
Tbf the one I linked is escompat not compat

**Comment by overlookmotel** [comment]
I stand corrected. That does make the impact of not supporting this less heavy. I still think we should though. Any friction for users is not what we want. And we also don't want to deal with triaging tons of issues saying "X plugin doesn't work"!

**Comment by connorshea** [comment]
Fixed by #18772
